File: src/test/java/no/nav/familie/ks/oppslag/config/RestTemplateConfig.java
package no.nav.familie.ks.oppslag.config;

import org.springframework.boot.web.client.RestTemplateBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import org.springframework.web.client.RestTemplate;

@Configuration
@Profile("dev")
public class RestTemplateConfig {

    @Bean
    public RestTemplate restTemplate() {
        return new RestTemplate();
    }

    @Bean
    public RestTemplate restTemplateMedProxy() {
        return new RestTemplateBuilder()
                .build();
    }

    @Bean
    public RestTemplateBuilder restTemplateBuilderMedProxy() {
        return new RestTemplateBuilder();
    }
}

File: src/main/java/no/nav/familie/integrasjoner/config/RestTemplateConfig.kt
package no.nav.familie.integrasjoner.config

import no.nav.familie.log.interceptor.ConsumerIdClientInterceptor
import no.nav.familie.log.interceptor.MdcValuesPropagatingClientInterceptor
import no.nav.familie.restklient.sts.StsBearerTokenClientInterceptor
import org.springframework.boot.web.client.RestTemplateBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestOperations
import java.time.Duration

/**
 * RestTemplate-konfig beholdt for SkyggesakRestClient (STS) og GraphQLWebClientConfig (TokenX).
 * Fjernes når disse klientene migreres/fjernes.
 */
@Configuration
class RestTemplateConfig {
    @Bean("sts")
    fun restTemplateSts(
        stsBearerTokenClientInterceptor: StsBearerTokenClientInterceptor,
        consumerIdClientInterceptor: ConsumerIdClientInterceptor,
    ): RestOperations =
        RestTemplateBuilder()
            .interceptors(
                consumerIdClientInterceptor,
                stsBearerTokenClientInterceptor,
                MdcValuesPropagatingClientInterceptor(),
            ).connectTimeout(Duration.ofSeconds(20))
            .readTimeout(Duration.ofSeconds(20))
            .build()
}
